src/main/java/com/cloudbees/plugins/deployer/engines/Engine.java

```diff
@@ -232,7 +234,7 @@ public void log(String message) {
         throw new DeployException("Deployment hosts of type " + configuration.getClass() + " are unsupported");
     }
 
-    public static class FingerprintDecorator implements Callable<DeployedApplicationLocation, IOException> {
+    public static class FingerprintDecorator extends MasterToSlaveCallable<DeployedApplicationLocation, IOException> {
```

Comment:
Last commit should make it. Please let me know if you think something is not appropriate.